### PR TITLE
Add query route primary data source when replica all offline.

### LINKED ISSUE #18441: Consider primary datasource provides query when all the replica are offline in Readwrite-Splitting Dynamic type
Hi community,

Execute the query throws an exception when we use `readwrite-splitting ` rule and` db-discovery` rule and all the replica are offline.

image

So, we consider add configuration switch in Dynamic `readwrite-splitting`, choose the primary data source by yourself to whether provide query when all the replica are offline.

### PR DESCRIPTION
**Pull request by zjcnb**
Fixes #18441

### CODE REVIEW

**Review by tristaZero — CHANGES_REQUESTED** [review]
How about `write-data-source-query-enable`?

**Comment by zjcnb** [comment]
> write-data-source-query-enable

Yeah, it 's better.

File: shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/route/impl/ReadwriteSplittingDataSourceRouter.java [hunk 1]
```diff
@@ -68,4 +69,9 @@ private boolean containsLastInsertIdProjection(final SQLStatementContext<?> sqlS
     private boolean isHintWriteRouteOnly(final SQLStatementContext<?> sqlStatementContext) {
         return HintManager.isWriteRouteOnly() || (sqlStatementContext instanceof CommonSQLStatementContext && ((CommonSQLStatementContext<?>) sqlStatementContext).isHintWriteRouteOnly());
     }
+    
+    private boolean isAllowWriteDatasourceQuery() {
```
Comment (menghaoranss) [inline shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/route/impl/ReadwriteSplittingDataSourceRouter.java:73]: Datasource should be DataSource

**Review by tristaZero — APPROVED** [review]

**Review by tristaZero — CHANGES_REQUESTED** [review]
test

**Review by tristaZero — CHANGES_REQUESTED** [review]
test

**Review by tristaZero — APPROVED** [review]

**Pull request merged**